File: lib/py/telepact/internal/binary/DefaultBinaryEncodingCache.py
from typing import Any, TYPE_CHECKING

from ...internal.binary.BinaryEncodingCache import BinaryEncodingCache

if TYPE_CHECKING:
    from .BinaryEncoding import BinaryEncoding

class DefaultBinaryEncodingCache(BinaryEncodingCache):

    def __init__(self) -> None:
        self.recent_binary_encoders: dict[int, 'BinaryEncoding'] = {}

    def add(self, checksum: int, binary_encoding_map: dict[str, int]) -> None:
        from .BinaryEncoding import BinaryEncoding        
        binary_encoding = BinaryEncoding(binary_encoding_map, checksum)
        self.recent_binary_encoders[checksum] = binary_encoding

    def get(self, checksum: int) -> 'BinaryEncoding':
        return self.recent_binary_encoders[checksum]

    def remove(self, checksum: int) -> None:
        self.recent_binary_encoders.pop(checksum, None)

File: lib/java/src/main/java/io/github/telepact/internal/binary/DefaultBinaryEncodingCache.java
package io.github.telepact.internal.binary;

import java.util.HashMap;
import java.util.Map;
import java.util.Optional;

public class DefaultBinaryEncodingCache implements BinaryEncodingCache {

    private final Map<Integer, BinaryEncoding> recentBinaryEncoders;

    public DefaultBinaryEncodingCache() {
        this.recentBinaryEncoders = new HashMap<>();
    }

    public void add(int checksum, Map<String, Integer> binaryEncodingMap) {
        BinaryEncoding binaryEncoding = new BinaryEncoding(binaryEncodingMap, checksum);
        this.recentBinaryEncoders.put(checksum, binaryEncoding);
    }

    public Optional<BinaryEncoding> get(int checksum) {
        if (!this.recentBinaryEncoders.containsKey(checksum)) {
            return Optional.empty();
        }
        return Optional.of(this.recentBinaryEncoders.get(checksum));
    }

    public void remove(int checksum) {
        this.recentBinaryEncoders.remove(checksum);
    }
}
